server/controllers/ShareController.js

```diff
@@ -210,6 +211,75 @@ class ShareController {
     res.sendFile(audioTrackPath)
   }
 
+  /**
+   * Public route
+   *
+   * GET: /api/share/:slug/download
+   * Downloads media item share
+   *
+   * @param {Request} req
+   * @param {Response} res
+   */
+  async downloadMediaItemShare(req, res) {
+    const { slug } = req.params
+
+    // Find matching MediaItemShare based on slug
+    const mediaItemShare = await ShareManager.findBySlug(slug)
+    // If the file isDownloadable, download the file
+    if (mediaItemShare.isDownloadable) {
+      // Get mediaItemId and type
+      const { mediaItemId, mediaItemType } = mediaItemShare
+
+      // Get the library item from the mediaItemId
+      const libraryItem = await Database.libraryItemModel.findOne({
+        where: {
+          mediaId: mediaItemId
```

Comment:
This isn't going to work when podcast episodes get supported because `mediaId` would be the podcast id and not the podcast episode id.
This is why I distinguish between `mediaId` and `mediaItemId`.

This works for books though since `mediaId` and `mediaItemId` is the same.

I'll update this part